src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -2609,22 +2599,38 @@ public void onFailure(Throwable caught) {
   }
 
   private void onRename() {
-    checkUpdateEntity(() -> {
-      postCheckRename();
-    });
+    if (checkIsLatestVersion()) {
```

Comment:
Can you add a test to verify that the error message is shown for datasets, and consider adding tests for any other cases that were missed (such as renaming a folder if your change caused that to break without a test failing)